src/attribution/onINP.ts

```diff
@@ -273,7 +280,14 @@ const attributeINP = (metric: INPMetric): INPMetricWithAttribution => {
     longAnimationFrameEntries.map((loaf) => loaf.startTime + loaf.duration),
   );
 
-  const nextPaintTime = Math.max.apply(Math, nextPaintTimeCandidates);
+  // processingEnd can extend beyond duration for modals where we artificially
+  // mark event timing duration as that paint.
+  // See: https://github.com/GoogleChrome/web-vitals/issues/492
+  // So cap to the INP value.
+  const nextPaintTime = Math.min(
```

Comment:
I'm not sure about this change...

Looking at like 276 above, you are already including `startTime + duration` in the list of paint time candidates, which means its in the list of MAX and then also in the list of MIN.  I think that means its always going to be picked?

(Note: I assume there is no difference between `firstEntry.duration` and `metric.value` but there can be)

Second, the comment anove is:

```js
// Since entry durations are rounded to the nearest 8ms, we need to clamp
// the `nextPaintTime` value to be higher than the `processingEnd` or
// end time of any LoAF entry.
```

I think we are specifically looking for cases where `startTime + duration` is "accidentally" less than one of the alternative paintTimeCandidates (such as processingEnd) when we do rounding.

Now here by using `startTime + duration` when it is less than the alternatives, we are effectively just undoing that?

Might as well just delete this code instead? Or, selectively do it if its "less than 8ms off" or something?

---

Finally, also already capped the processingEnd time in this CL, so its also not useful to look at processingEnd as an alternative to paint time any more (it will always be smaller now).

Reply:
> Now here by using startTime + duration when it is less than the alternatives, we are effectively just undoing that?

Urgh! I realised this earlier, and then completely forgot about it and committed it with further.

> Might as well just delete this code instead?

I think we need to do _something_ here, or we'll just move this incorrect breakdown to presentation delay.

> Or, selectively do it if its "less than 8ms off" or something?

Yeah think we need something like this. Or if `processingEnd` is `firstEntry.startTime + metric.value`, then we can 0 out the presentation time.

Reply:
Have another look not if you can @mmocny